### ISSUE TITLE
 0.90.91 - does not work properly with external keyboard

### ISSUE THREAD

**Issue by leipreachan** [tekezo/Karabiner-Elements#662]
I have remapped Caps Lock to F19 and F19 is selected to "Select previous input source".
This has been working fine for a long time.
Today I've updated to 0.90.91 and the external keyboard doesn't get this mapping. It doesn't switch the layout. Turned it off and everything is ok. 

Please find the logs below:
```
[2017-03-15 11:13:17.485] [info] [grabber] version 0.90.91
[2017-03-15 11:13:17.486] [info] [grabber] Check version...
[2017-03-15 11:13:17.666] [info] [grabber] IOServiceOpen is succeeded @ void krbn::virtual_hid_device_client::matched_callback(io_iterator_t)
[2017-03-15 11:13:17.671] [info] [grabber] matching device:
[2017-03-15 11:13:17.671] [info] [grabber]   manufacturer: Apple Inc.
[2017-03-15 11:13:17.671] [info] [grabber]   product: Apple Internal Keyboard / Trackpad
[2017-03-15 11:13:17.671] [info] [grabber]   vendor_id: 0x5ac
[2017-03-15 11:13:17.671] [info] [grabber]   product_id: 0x274
[2017-03-15 11:13:17.671] [info] [grabber]   location_id: 0x14400000
[2017-03-15 11:13:17.672] [info] [grabber]   serial_number: D3H62745FT1FTV4AK5FF
[2017-03-15 11:13:17.672] [info] [grabber]   registry_entry_id: 4294968160
[2017-03-15 11:13:17.672] [info] [console_user_server] connection_manager::grabber_is_launched_callback
[2017-03-15 11:13:17.673] [info] [grabber] matching device:
[2017-03-15 11:13:17.673] [info] [grabber]   manufacturer: Apple
[2017-03-15 11:13:17.673] [info] [grabber]   product: Keyboard
[2017-03-15 11:13:17.673] [info] [grabber]   vendor_id: 0x5ac
[2017-03-15 11:13:17.673] [info] [grabber]   product_id: 0xffff
[2017-03-15 11:13:17.673] [info] [grabber]   registry_entry_id: 4294967997
[2017-03-15 11:13:17.676] [info] [grabber] matching device:
[2017-03-15 11:13:17.676] [info] [grabber]   manufacturer: Apple
[2017-03-15 11:13:17.676] [info] [grabber]   product: Keyboard
[2017-03-15 11:13:17.676] [info] [grabber]   vendor_id: 0x5ac
[2017-03-15 11:13:17.676] [info] [grabber]   product_id: 0xffff
[2017-03-15 11:13:17.676] [info] [grabber]   registry_entry_id: 4294967999
[2017-03-15 11:13:17.819] [info] [console_user_server] Check version...
[2017-03-15 11:13:17.819] [warning] [console_user_server] grabber socket is not found
[2017-03-15 11:13:18.723] [info] [grabber] current_console_user_id: 502
[2017-03-15 11:13:18.723] [info] [grabber] Check version...
[2017-03-15 11:13:18.726] [info] [grabber] event_tap_manager initialized
[2017-03-15 11:13:18.726] [info] [grabber] Load karabiner.json...
[2017-03-15 11:13:18.726] [info] [grabber] core_configuration is updated.
[2017-03-15 11:13:18.726] [info] [grabber] initialize_virtual_hid_keyboard
[2017-03-15 11:13:18.726] [info] [grabber]   keyboard_type:41
[2017-03-15 11:13:18.726] [info] [grabber]   caps_lock_delay_milliseconds:0
[2017-03-15 11:13:18.727] [info] [grabber] Load karabiner.json...
[2017-03-15 11:13:18.728] [info] [grabber] core_configuration is updated.
[2017-03-15 11:13:18.729] [info] [grabber] matching device:
[2017-03-15 11:13:18.729] [info] [grabber]   manufacturer: pqrs.org
[2017-03-15 11:13:18.729] [info] [grabber]   product: Karabiner VirtualHIDKeyboard
[2017-03-15 11:13:18.729] [info] [grabber]   vendor_id: 0x0
[2017-03-15 11:13:18.729] [info] [grabber]   product_id: 0x0
[2017-03-15 11:13:18.729] [info] [grabber]   location_id: 0x0
[2017-03-15 11:13:18.729] [info] [grabber]   serial_number: org.pqrs.driver.Karabiner.VirtualHIDDevice.VirtualHIDKeyboard
[2017-03-15 11:13:18.729] [info] [grabber]   registry_entry_id: 4294970574
[2017-03-15 11:13:18.770] [info] [grabber] karabiner_console_user_server is connected (pid:1298)
[2017-03-15 11:13:18.770] [info] [grabber] event_tap_manager initialized
[2017-03-15 11:13:18.770] [info] [grabber] event_tap_manager terminated
[2017-03-15 11:13:18.770] [info] [grabber] Load karabiner.json...
[2017-03-15 11:13:18.770] [info] [grabber] core_configuration is updated.
[2017-03-15 11:13:18.770] [info] [grabber] initialize_virtual_hid_keyboard
[2017-03-15 11:13:18.770] [info] [grabber]   keyboard_type:40
[2017-03-15 11:13:18.770] [info] [grabber]   caps_lock_delay_milliseconds:0
[2017-03-15 11:13:18.770] [info] [console_user_server] Check version...
[2017-03-15 11:13:18.770] [info] [console_user_server] grabber_client_ is connected
[2017-03-15 11:13:18.770] [info] [console_user_server] Load karabiner.json...
[2017-03-15 11:13:18.770] [info] [console_user_server] core_configuration is updated.
[2017-03-15 11:13:18.770] [info] [console_user_server] Check for updates...
[2017-03-15 11:13:18.772] [info] [grabber] Load karabiner.json...
[2017-03-15 11:13:18.772] [info] [grabber] core_configuration is updated.
[2017-03-15 11:13:18.773] [info] [grabber] removal device:
[2017-03-15 11:13:18.773] [info] [grabber]   vendor_id: 0x0
[2017-03-15 11:13:18.773] [info] [grabber]   product_id: 0x0
[2017-03-15 11:13:18.773] [info] [grabber]   location_id: 0x0
[2017-03-15 11:13:18.779] [info] [grabber] matching device:
[2017-03-15 11:13:18.779] [info] [grabber]   manufacturer: pqrs.org
[2017-03-15 11:13:18.780] [info] [grabber]   product: Karabiner VirtualHIDKeyboard
[2017-03-15 11:13:18.780] [info] [grabber]   vendor_id: 0x0
[2017-03-15 11:13:18.780] [info] [grabber]   product_id: 0x0
[2017-03-15 11:13:18.780] [info] [grabber]   location_id: 0x0
[2017-03-15 11:13:18.780] [info] [grabber]   serial_number: org.pqrs.driver.Karabiner.VirtualHIDDevice.VirtualHIDKeyboard
[2017-03-15 11:13:18.780] [info] [grabber]   registry_entry_id: 4294970587
[2017-03-15 11:13:18.893] [info] [console_user_server] Load karabiner.json...
[2017-03-15 11:13:18.893] [info] [console_user_server] core_configuration is updated.
[2017-03-15 11:13:18.945] [info] [grabber] Keyboard is grabbed
[2017-03-15 11:13:18.945] [info] [grabber] Keyboard is grabbed
[2017-03-15 11:13:18.978] [info] [grabber] Apple Internal Keyboard / Trackpad is grabbed
[2017-03-15 11:13:19.834] [info] [grabber] system_preferences_values_updated
[2017-03-15 11:13:19.834] [info] [console_user_server] system_preferences::values is updated.
[2017-03-15 13:05:39.292] [info] [grabber] matching device:
[2017-03-15 13:05:39.292] [info] [grabber]   manufacturer: Logitech
[2017-03-15 13:05:39.292] [info] [grabber]   product: USB Receiver
[2017-03-15 13:05:39.292] [info] [grabber]   vendor_id: 0x46d
[2017-03-15 13:05:39.292] [info] [grabber]   product_id: 0xc52b
[2017-03-15 13:05:39.292] [info] [grabber]   location_id: 0x14100000
[2017-03-15 13:05:39.292] [info] [grabber]   registry_entry_id: 4294971377
[2017-03-15 13:05:39.395] [info] [grabber] USB Receiver is grabbed
[2017-03-15 13:05:44.386] [info] [grabber] matching device:
[2017-03-15 13:05:44.386] [info] [grabber]   manufacturer: Apple
[2017-03-15 13:05:44.386] [info] [grabber]   product: Unifying Device
[2017-03-15 13:05:44.386] [info] [grabber]   vendor_id: 0x5ac
[2017-03-15 13:05:44.387] [info] [grabber]   product_id: 0xffff
[2017-03-15 13:05:44.387] [info] [grabber]   location_id: 0x14100100
[2017-03-15 13:05:44.387] [info] [grabber]   registry_entry_id: 4294971438
[2017-03-15 13:05:44.417] [info] [grabber] matching device:
[2017-03-15 13:05:44.417] [info] [grabber]   manufacturer: Apple
[2017-03-15 13:05:44.417] [info] [grabber]   product: Unifying Device
[2017-03-15 13:05:44.417] [info] [grabber]   vendor_id: 0x5ac
[2017-03-15 13:05:44.417] [info] [grabber]   product_id: 0xffff
[2017-03-15 13:05:44.417] [info] [grabber]   location_id: 0x14100100
[2017-03-15 13:05:44.417] [info] [grabber]   registry_entry_id: 4294971445
[2017-03-15 13:05:44.427] [info] [grabber] matching device:
[2017-03-15 13:05:44.427] [info] [grabber]   manufacturer: Apple
[2017-03-15 13:05:44.427] [info] [grabber]   product: Unifying Device
[2017-03-15 13:05:44.427] [info] [grabber]   vendor_id: 0x5ac
[2017-03-15 13:05:44.427] [info] [grabber]   product_id: 0xffff
[2017-03-15 13:05:44.427] [info] [grabber]   location_id: 0x14100200
[2017-03-15 13:05:44.427] [info] [grabber]   registry_entry_id: 4294971472
[2017-03-15 13:05:45.739] [info] [grabber] version 0.90.91
[2017-03-15 13:05:45.740] [info] [grabber] Check version...
[2017-03-15 13:05:45.955] [info] [grabber] IOServiceOpen is succeeded @ void krbn::virtual_hid_device_client::matched_callback(io_iterator_t)
[2017-03-15 13:05:45.962] [info] [grabber] matching device:
[2017-03-15 13:05:45.962] [info] [grabber]   manufacturer: Logitech
[2017-03-15 13:05:45.962] [info] [grabber]   product: Unifying Device
[2017-03-15 13:05:45.962] [info] [grabber]   vendor_id: 0x46d
[2017-03-15 13:05:45.963] [info] [grabber]   product_id: 0x4002
[2017-03-15 13:05:45.963] [info] [grabber]   location_id: 0x14100200
[2017-03-15 13:05:45.963] [info] [grabber]   registry_entry_id: 4294971432
[2017-03-15 13:05:45.963] [info] [console_user_server] connection_manager::grabber_is_launched_callback
[2017-03-15 13:05:45.966] [info] [grabber] matching device:
[2017-03-15 13:05:45.967] [info] [grabber]   manufacturer: Apple Inc.
[2017-03-15 13:05:45.967] [info] [grabber]   product: Apple Internal Keyboard / Trackpad
[2017-03-15 13:05:45.967] [info] [grabber]   vendor_id: 0x5ac
[2017-03-15 13:05:45.967] [info] [grabber]   product_id: 0x274
[2017-03-15 13:05:45.967] [info] [grabber]   location_id: 0x14400000
[2017-03-15 13:05:45.967] [info] [grabber]   serial_number: D3H62745FT1FTV4AK5FF
[2017-03-15 13:05:45.967] [info] [grabber]   registry_entry_id: 4294968160
[2017-03-15 13:05:45.969] [info] [grabber] matching device:
[2017-03-15 13:05:45.969] [info] [grabber]   manufacturer: Apple
[2017-03-15 13:05:45.969] [info] [grabber]   product: Unifying Device
[2017-03-15 13:05:45.969] [info] [grabber]   vendor_id: 0x5ac
[2017-03-15 13:05:45.969] [info] [grabber]   product_id: 0xffff
[2017-03-15 13:05:45.969] [info] [grabber]   location_id: 0x14100200
[2017-03-15 13:05:45.969] [info] [grabber]   registry_entry_id: 4294971472
[2017-03-15 13:05:45.971] [info] [grabber] matching device:
[2017-03-15 13:05:45.971] [info] [grabber]   manufacturer: Apple
[2017-03-15 13:05:45.971] [info] [grabber]   product: Unifying Device
[2017-03-15 13:05:45.971] [info] [grabber]   vendor_id: 0x5ac
[2017-03-15 13:05:45.971] [info] [grabber]   product_id: 0xffff
[2017-03-15 13:05:45.971] [info] [grabber]   location_id: 0x14100100
[2017-03-15 13:05:45.971] [info] [grabber]   registry_entry_id: 4294971438
[2017-03-15 13:05:45.974] [info] [grabber] matching device:
[2017-03-15 13:05:45.974] [info] [grabber]   manufacturer: Logitech
[2017-03-15 13:05:45.974] [info] [grabber]   product: USB Receiver
[2017-03-15 13:05:45.974] [info] [grabber]   vendor_id: 0x46d
[2017-03-15 13:05:45.974] [info] [grabber]   product_id: 0xc52b
[2017-03-15 13:05:45.974] [info] [grabber]   location_id: 0x14100000
[2017-03-15 13:05:45.974] [info] [grabber]   registry_entry_id: 4294971377
[2017-03-15 13:05:45.975] [info] [grabber] matching device:
[2017-03-15 13:05:45.975] [info] [grabber]   manufacturer: Apple
[2017-03-15 13:05:45.975] [info] [grabber]   product: Unifying Device
[2017-03-15 13:05:45.975] [info] [grabber]   vendor_id: 0x5ac
[2017-03-15 13:05:45.975] [info] [grabber]   product_id: 0xffff
[2017-03-15 13:05:45.976] [info] [grabber]   location_id: 0x14100200
[2017-03-15 13:05:45.976] [info] [grabber]   registry_entry_id: 4294971467
[2017-03-15 13:05:45.978] [info] [grabber] matching device:
[2017-03-15 13:05:45.978] [info] [grabber]   manufacturer: Apple
[2017-03-15 13:05:45.978] [info] [grabber]   product: Keyboard
[2017-03-15 13:05:45.978] [info] [grabber]   vendor_id: 0x5ac
[2017-03-15 13:05:45.978] [info] [grabber]   product_id: 0xffff
[2017-03-15 13:05:45.979] [info] [grabber]   registry_entry_id: 4294967997
[2017-03-15 13:05:45.981] [info] [grabber] matching device:
[2017-03-15 13:05:45.981] [info] [grabber]   manufacturer: Apple
[2017-03-15 13:05:45.981] [info] [grabber]   product: Keyboard
[2017-03-15 13:05:45.981] [info] [grabber]   vendor_id: 0x5ac
[2017-03-15 13:05:45.981] [info] [grabber]   product_id: 0xffff
[2017-03-15 13:05:45.981] [info] [grabber]   registry_entry_id: 4294967999
[2017-03-15 13:05:46.844] [info] [console_user_server] Check version...
[2017-03-15 13:05:46.844] [warning] [console_user_server] grabber socket is not found
[2017-03-15 13:05:47.845] [info] [console_user_server] Check version...
[2017-03-15 13:05:47.845] [warning] [console_user_server] grabber socket is not found
[2017-03-15 13:05:48.770] [info] [console_user_server] Check version...
[2017-03-15 13:05:48.770] [warning] [console_user_server] grabber socket is not found
[2017-03-15 13:05:49.813] [info] [console_user_server] Check version...
[2017-03-15 13:05:49.813] [warning] [console_user_server] grabber socket is not found
[2017-03-15 13:05:50.844] [info] [console_user_server] Check version...
[2017-03-15 13:05:50.844] [warning] [console_user_server] grabber socket is not found
[2017-03-15 13:05:51.813] [info] [console_user_server] Check version...
[2017-03-15 13:05:51.813] [warning] [console_user_server] grabber socket is not found
[2017-03-15 13:05:52.844] [info] [console_user_server] Check version...
[2017-03-15 13:05:52.844] [warning] [console_user_server] grabber socket is not found
[2017-03-15 13:05:53.770] [info] [console_user_server] Check version...
[2017-03-15 13:05:53.770] [warning] [console_user_server] grabber socket is not found
[2017-03-15 13:05:54.770] [info] [console_user_server] Check version...
[2017-03-15 13:05:54.770] [warning] [console_user_server] grabber socket is not found
[2017-03-15 13:05:55.771] [info] [console_user_server] Check version...
[2017-03-15 13:05:55.771] [warning] [console_user_server] grabber socket is not found
[2017-03-15 13:05:56.170] [info] [grabber] version 0.90.91
[2017-03-15 13:05:56.171] [info] [grabber] Check version...
[2017-03-15 13:05:56.371] [info] [grabber] IOServiceOpen is succeeded @ void krbn::virtual_hid_device_client::matched_callback(io_iterator_t)
[2017-03-15 13:05:56.376] [info] [grabber] matching device:
[2017-03-15 13:05:56.377] [info] [grabber]   manufacturer: Apple
[2017-03-15 13:05:56.377] [info] [grabber]   product: Unifying Device
[2017-03-15 13:05:56.377] [info] [grabber]   vendor_id: 0x5ac
[2017-03-15 13:05:56.377] [info] [grabber]   product_id: 0xffff
[2017-03-15 13:05:56.377] [info] [grabber]   location_id: 0x14100200
[2017-03-15 13:05:56.377] [info] [grabber]   registry_entry_id: 4294971472
[2017-03-15 13:05:56.377] [info] [console_user_server] connection_manager::grabber_is_launched_callback
[2017-03-15 13:05:56.379] [info] [grabber] matching device:
[2017-03-15 13:05:56.379] [info] [grabber]   manufacturer: Apple Inc.
[2017-03-15 13:05:56.379] [info] [grabber]   product: Apple Internal Keyboard / Trackpad
[2017-03-15 13:05:56.379] [info] [grabber]   vendor_id: 0x5ac
[2017-03-15 13:05:56.379] [info] [grabber]   product_id: 0x274
[2017-03-15 13:05:56.379] [info] [grabber]   location_id: 0x14400000
[2017-03-15 13:05:56.379] [info] [grabber]   serial_number: D3H62745FT1FTV4AK5FF
[2017-03-15 13:05:56.379] [info] [grabber]   registry_entry_id: 4294968160
[2017-03-15 13:05:56.381] [info] [grabber] matching device:
[2017-03-15 13:05:56.381] [info] [grabber]   manufacturer: Logitech
[2017-03-15 13:05:56.381] [info] [grabber]   product: Unifying Device
[2017-03-15 13:05:56.381] [info] [grabber]   vendor_id: 0x46d
[2017-03-15 13:05:56.381] [info] [grabber]   product_id: 0x4002
[2017-03-15 13:05:56.381] [info] [grabber]   location_id: 0x14100200
[2017-03-15 13:05:56.381] [info] [grabber]   registry_entry_id: 4294971432
[2017-03-15 13:05:56.385] [info] [grabber] matching device:
[2017-03-15 13:05:56.385] [info] [grabber]   manufacturer: Apple
[2017-03-15 13:05:56.385] [info] [grabber]   product: Unifying Device
[2017-03-15 13:05:56.385] [info] [grabber]   vendor_id: 0x5ac
[2017-03-15 13:05:56.385] [info] [grabber]   product_id: 0xffff
[2017-03-15 13:05:56.385] [info] [grabber]   location_id: 0x14100200
[2017-03-15 13:05:56.385] [info] [grabber]   registry_entry_id: 4294971467
[2017-03-15 13:05:56.388] [info] [grabber] matching device:
[2017-03-15 13:05:56.388] [info] [grabber]   manufacturer: Apple
[2017-03-15 13:05:56.388] [info] [grabber]   product: Keyboard
[2017-03-15 13:05:56.388] [info] [grabber]   vendor_id: 0x5ac
[2017-03-15 13:05:56.388] [info] [grabber]   product_id: 0xffff
[2017-03-15 13:05:56.388] [info] [grabber]   registry_entry_id: 4294967999
[2017-03-15 13:05:56.391] [info] [grabber] matching device:
[2017-03-15 13:05:56.391] [info] [grabber]   manufacturer: Apple
[2017-03-15 13:05:56.391] [info] [grabber]   product: Unifying Device
[2017-03-15 13:05:56.391] [info] [grabber]   vendor_id: 0x5ac
[2017-03-15 13:05:56.391] [info] [grabber]   product_id: 0xffff
[2017-03-15 13:05:56.391] [info] [grabber]   location_id: 0x14100100
[2017-03-15 13:05:56.391] [info] [grabber]   registry_entry_id: 4294971445
[2017-03-15 13:05:56.394] [info] [grabber] matching device:
[2017-03-15 13:05:56.394] [info] [grabber]   manufacturer: Apple
[2017-03-15 13:05:56.394] [info] [grabber]   product: Unifying Device
[2017-03-15 13:05:56.394] [info] [grabber]   vendor_id: 0x5ac
[2017-03-15 13:05:56.394] [info] [grabber]   product_id: 0xffff
[2017-03-15 13:05:56.394] [info] [grabber]   location_id: 0x14100100
[2017-03-15 13:05:56.394] [info] [grabber]   registry_entry_id: 4294971438
[2017-03-15 13:05:56.770] [info] [console_user_server] Check version...
[2017-03-15 13:05:56.770] [warning] [console_user_server] grabber socket is not found
[2017-03-15 13:05:57.844] [info] [console_user_server] Check version...
[2017-03-15 13:05:57.844] [warning] [console_user_server] grabber socket is not found
[2017-03-15 13:05:58.814] [info] [console_user_server] Check version...
[2017-03-15 13:05:58.814] [warning] [console_user_server] grabber socket is not found
[2017-03-15 13:05:59.844] [info] [console_user_server] Check version...
[2017-03-15 13:05:59.844] [warning] [console_user_server] grabber socket is not found
[2017-03-15 13:06:00.814] [info] [console_user_server] Check version...
[2017-03-15 13:06:00.814] [warning] [console_user_server] grabber socket is not found
[2017-03-15 13:06:01.843] [info] [console_user_server] Check version...
[2017-03-15 13:06:01.843] [warning] [console_user_server] grabber socket is not found
[2017-03-15 13:06:02.770] [info] [console_user_server] Check version...
[2017-03-15 13:06:02.770] [warning] [console_user_server] grabber socket is not found
[2017-03-15 13:06:03.844] [info] [console_user_server] Check version...
[2017-03-15 13:06:03.844] [warning] [console_user_server] grabber socket is not found
[2017-03-15 13:06:04.844] [info] [console_user_server] Check version...
[2017-03-15 13:06:04.844] [warning] [console_user_server] grabber socket is not found
[2017-03-15 13:06:05.770] [info] [console_user_server] Check version...
```

Sierra 10.12.3

**Comment by studgeek** [comment]
I have the same problem.

**Comment by dboyd68** [comment]
+1 same issue

**Comment by 2t10p** [comment]
same issue 
.
and if you need old version 
can download from here
https://github.com/tekezo/pqrs.org/tree/master/webroot/osx/karabiner/files

**Comment by leipreachan** [comment]
btw I tried the old version.
Even with the old version I have input lags with external logitech keyboard / mouse.
I turn Karabiner off and no lags at all.
I'm sad, Karabiner is a wonderful tool.

**Comment by tekezo** [comment]
Sorry for inconvenience.
I've fixed this issue at v0.90.92.
Please upgrade Karabiner-Elements.
